packages/design-system-react-native/src/components/ButtonIcon/ButtonIcon.types.ts

```diff
@@ -0,0 +1,62 @@
+import type { PressableProps, StyleProp, ViewStyle } from 'react-native';
+
+import type { IconProps, IconName } from '../Icon';
+
+export enum ButtonIconSize {
+  /**
+   * Represents a small button size (24px).
+   */
+  Sm = '24',
+  /**
+   * Represents a medium button size (32px).
+   */
+  Md = '32',
+  /**
+   * Represents a large button size (40px).
+   */
+  Lg = '40',
+}
+
+/**
+ * ButtonIcon component props.
+ */
+export type ButtonIconProps = {
+  /**
+   * Optional prop to control the size of the icon
+   * Different sizes map to specific pixel dimensions
+   * @default IconSize.Md
+   */
+  size?: ButtonIconSize;
+  /**
+   * Optional prop to specify an icon to show
+   */
+  iconName: IconName;
+  /**
+   * Optional prop to pass additional properties to the icon
+   */
+  iconProps?: Partial<IconProps>;
+  /**
+   * Optional prop that when true, disables the button
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Optional prop to show the inverse state of the button, which is reserved for buttons on colored backgrounds.
+   * @default false
+   */
+  isInverse?: boolean;
+  /**
+   * Optional prop to show the floating/contained state of the button, which is reserved for floating buttons.
+   * Note: This prop provides styling only. There is no positioning logic.
+   * @default false
+   */
+  isFloating?: boolean;
+  /**
+   * Optional prop to add twrnc overriding classNames.
+   */
+  twClassName?: string;
+  /**
+   * Optional prop to control the style.
+   */
+  style?: StyleProp<ViewStyle>;
+} & Omit<PressableProps, 'children'>;
```

Comment:
possibly blocking: I noticed we don't have a required `accessibilityLabel` prop for the icon. I'm not very familiar with accessibility best practices for mobile, but I think labeling icon buttons is important. How should we handle the equivalent of `aria-label` in React Native, and how should it be applied here?

Reply:
this extends `PressableProps`, which inherits `accessibilityLabel` 
<img width="235" alt="Screenshot 2025-02-04 at 9 52 07 AM" src="https://github.com/user-attachments/assets/365307a1-5dad-4057-a34d-64d9435715de" />
<img width="607" alt="Screenshot 2025-02-04 at 9 52 14 AM" src="https://github.com/user-attachments/assets/d3775cec-55ed-4d84-8937-3dde33248259" />

Reply:
Should `accessibilityLabel` or `aria-label` be a required prop in `ButtonIcon`?

Reply:
As discussed let's merge and add accessibility props in another PR. Issue here: https://github.com/MetaMask/metamask-design-system/issues/421